firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java

```diff
@@ -691,10 +691,17 @@ public QueryTarget encodeQueryTarget(Query query) {
     // Dissect the path into parent, collectionId, and optional key filter.
     QueryTarget.Builder builder = QueryTarget.newBuilder();
     StructuredQuery.Builder structuredQueryBuilder = StructuredQuery.newBuilder();
-    if (query.getPath().length() == 0) {
-      builder.setParent(encodeQueryPath(ResourcePath.EMPTY));
+    ResourcePath path = query.getPath();
+    if (query.getCollectionGroup() != null) {
+      Assert.hardAssert(
+          path.length() % 2 == 0,
+          "Collection Group queries should be within a document path or root.");
+      builder.setParent(encodeQueryPath(path));
```

Comment:
Optional: add a newline before this line?

Reply:
Opted not to add this one.